### DISCUSSION TITLE
ODoH messages paddings

### DISCUSSION THREAD

**Discussion by AmirSalari** [DNSCrypt/dnscrypt-proxy#2271] [Q&A]
Hi,
I wanted to ask if ODoH messages are padded in its implementation in the dnscrypt-proxy. In ODoH [RFC ](https://datatracker.ietf.org/doc/rfc9230/), in client implementation it is mentioned that:

> Oblivious Client Behavior
> 
>    Let M be a DNS message (query) a Client wishes to protect with
>    Oblivious DoH.  When sending an Oblivious DoH Query for resolving M
>    to an Oblivious Target with ObliviousDoHConfigContents config, a
>    Client does the following:
> 
>    **1.  Creates an ObliviousDoHQuery structure, carrying the message M
>        and padding, to produce Q_plain.**
>

**Comment by jedisct1** [comment]
Both `dnscrypt-proxy` and `doh-server` unconditionally add padding to the DNS queries and responses.

So, at the ODoH layer, they both don't add any extra padding, as that would be redundant.
